### PR TITLE
Support for creating snapshots using ark

### PR DESCRIPTION
**Pull request by disrani-px**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+ci-%:
+	$(MAKE) --no-print-directory BIN=$* ci
+
+ci:
+	mkdir -p _output
+	CGO_ENABLED=0 go build -v -o _output/$(BIN) ./$(BIN)
+
+clean:
+	rm -rf .go _output
```
Comment (jvinod) [inline Makefile:88]: suggest adding vet and lint targets right from the beginning:
```

ifndef PKGS
PKGS := $(shell go list ./... 2>&1 | grep -v 'vendor' | grep -v 'sanity')
endif

lint:
	go get -v github.com/golang/lint/golint
	golint $(PKGS)

vet:
	go vet $(PKGS)
```

File: pkg/snapshot/localsnap.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	if err != nil {
+		return "", err
+	}
+	if len(vols) == 0 {
+		return "", fmt.Errorf("Volume %v not found", volumeID)
+	}
+
+	tags["pvName"] = vols[0].Locator.Name
+	locator := &api.VolumeLocator{
+		Name:         tags["ark.heptio.com/backup"] + "_" + vols[0].Locator.Name,
```
Comment (jvinod) [inline pkg/snapshot/localsnap.go:82]: Is there a conflict - if called twice

File: pkg/snapshot/localsnap.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	if err != nil {
+		return false, err
+	}
+	if len(vols) == 0 {
+		return false, fmt.Errorf("Volume %v not found", volumeID)
+	}
+	return true, nil
+}
+
+func (l *localSnapshotPlugin) CreateSnapshot(volumeID, volumeAZ string, tags map[string]string) (string, error) {
```
Comment (jvinod) [inline pkg/snapshot/localsnap.go:66]: So no group snapshots in this api? Should we propose that to them?  Can tags be used to group volumes?

**Review by jvinod — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (disrani-px) [reply Makefile:88]: Yup forgot about those. Added vet, lint and errcheck.

File: pkg/snapshot/localsnap.go (hunk 2 above)
Comment (disrani-px) [reply pkg/snapshot/localsnap.go:82]: The backup names need to be unique from ark.

File: pkg/snapshot/localsnap.go (hunk 3 above)
Comment (disrani-px) [reply pkg/snapshot/localsnap.go:66]: Yes, I already proposed it to them. They have raised a feature bug: https://github.com/heptio/ark/issues/505

**Pull request merged**